OpenDreamRuntime/Objects/Types/DreamObjectSavefile.cs

```diff
@@ -124,9 +124,22 @@ public sealed class DreamObjectSavefile : DreamObject {
         Savefiles.Add(this);
     }
 
-    protected override void HandleDeletion() {
+    protected override void HandleDeletion(bool possiblyThreaded) {
+        // SAFETY: Close() is not threadsafe and doesn't have reason to be.
+        if (possiblyThreaded) {
+            EnterIntoDelQueue();
+            return;
+        }
+
+        if (Deleted)
+            return;
```

Comment:
`DreamObject.Delete()` should already be checking this before calling `HandleDeletion()`